src/syrupy/report.py

```diff
@@ -36,8 +36,8 @@
 @attr.s
 class SnapshotReport:
     base_dir: str = attr.ib()
-    all_items: Set[Any] = attr.ib()
-    ran_items: Set[Any] = attr.ib()
+    all_items: Dict["pytest.Item", bool] = attr.ib()
```

Comment:
changed this to a Dict to ensure iteration in deterministic order. Shouldn't affect performance, and the deterministic order allows us to properly test this bug fix